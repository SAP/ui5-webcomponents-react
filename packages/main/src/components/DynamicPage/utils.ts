export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
