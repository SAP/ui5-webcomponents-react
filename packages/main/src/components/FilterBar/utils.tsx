import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
