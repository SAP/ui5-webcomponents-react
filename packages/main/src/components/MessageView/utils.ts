import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
