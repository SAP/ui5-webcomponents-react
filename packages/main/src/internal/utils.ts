import { Children, Fragment, ReactNode } from 'react';

export function flattenFragments(children: ReactNode | ReactNode[], depth = 1): ReactNode[] {
  const flatChildren = [];
  if (Array.isArray(children)) {
    return children;
  }
  const removeFragments = (element, level = 0) => {
    if (!element) {
      return;
    }
    if (element.type === Fragment && level <= depth) {
      Children.toArray(element.props?.children).forEach((item) => {
        removeFragments(item, level + 1);
      });
    } else {
      flatChildren.push(element);
    }
  };
  removeFragments(children);

  return flatChildren;
}
