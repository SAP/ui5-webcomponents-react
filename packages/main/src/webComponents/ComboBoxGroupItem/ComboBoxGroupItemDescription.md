The `ComboBoxGroupItem` is type of suggestion item, that can be used to split the `ComboBox` suggestions into groups
