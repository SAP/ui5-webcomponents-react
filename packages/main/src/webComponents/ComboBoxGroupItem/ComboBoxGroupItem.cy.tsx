import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { ComboBoxGroupItem } from './index';

describe('ComboBoxGroupItem', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<ComboBoxGroupItem />);
    cy.get('[ui5-cb-group-item]').should('exist');
  });

  cypressPassThroughTestsFactory(ComboBoxGroupItem);
});
