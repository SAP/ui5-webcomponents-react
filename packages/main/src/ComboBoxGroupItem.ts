import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { ComboBoxGroupItemDomRef, ComboBoxGroupItemPropTypes } from './webComponents/ComboBoxGroupItem';
import { ComboBoxGroupItem } from './webComponents/ComboBoxGroupItem';

deprecationNotice(
  'ComboBoxGroupItem',
  `importing from '@ui5/webcomponents-react/dist/ComboBoxGroupItem' is deprecated. Please use "import { ComboBoxGroupItem } from '@ui5/webcomponents-react';" instead.`
);

export { ComboBoxGroupItem };

export type { ComboBoxGroupItemPropTypes, ComboBoxGroupItemDomRef };
