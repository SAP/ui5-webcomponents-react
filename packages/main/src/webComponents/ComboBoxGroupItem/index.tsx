import { withWebComponent } from '@ui5/webcomponents-react/dist/withWebComponent';
import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';

import '@ui5/webcomponents/dist/ComboBoxGroupItem';

export interface ComboBoxGroupItemPropTypes extends CommonProps {
  /**
   * Defines the text of the component.
   */
  text?: string;
}

/**
 * The `undefined` is type of suggestion item, that can be used to split the `ComboBox` suggestions into groups
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/ComboBoxGroupItem" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const ComboBoxGroupItem = withWebComponent<ComboBoxGroupItemPropTypes>('ui5-cb-group-item', ['text'], [], [], []);

ComboBoxGroupItem.displayName = 'ComboBoxGroupItem';

ComboBoxGroupItem.defaultProps = {};

export { ComboBoxGroupItem };
