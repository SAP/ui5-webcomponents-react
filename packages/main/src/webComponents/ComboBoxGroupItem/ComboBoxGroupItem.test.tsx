import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { ComboBoxGroupItem } from '@ui5/webcomponents-react/dist/ComboBoxGroupItem';
import React from 'react';

describe('ComboBoxGroupItem', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<ComboBoxGroupItem />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(ComboBoxGroupItem);
});
