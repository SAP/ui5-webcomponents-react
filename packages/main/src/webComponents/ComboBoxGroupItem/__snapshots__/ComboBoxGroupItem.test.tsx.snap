// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboBoxGroupItem Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-cb-group-item
    ui5-cb-group-item=""
  />
</DocumentFragment>
`;
